api/common/async_requests.py

```diff
@@ -16,8 +16,21 @@ def __init__(self, response_dict: httpx.Response):
         self.status = response_dict.status_code
 
 
+async def retry_request(*args, **kwargs):
+    retries = 1
+    backoff_factor = 0.1
+    status_codes = {502, 503, 504}
+
+    for attempt in range(retries + 1):
+        response = await client.get(*args, **kwargs)
+        if response.status_code not in status_codes:
```

Comment:
This is going to rely on getting a response, but do we care about classes of error that don't return a valid response, like a connection error? I raise this because I don't think it replicates the change you just made in https://github.com/DemocracyClub/UK-Polling-Stations/pull/8487/commits/c980772faf865309b21ca3dc7816a21fed18adf6

Reply:
Yeah, seen the next commit. Nothing to see here